src/raftstore/store/peer.rs

```diff
@@ -1349,20 +1349,28 @@ impl Peer {
                 // have no effect.
                 self.proposals.clear();
             }
+            if lease_to_be_updated {
```

Comment:
can we check all in the above same `for` loop ?

Reply:
We can but the implement will be odd, because `find_propose_time` will clear proposals with lower index or lower term until find the correct proposal or proposal with higher term. Here, we need to processing committed entry from lower index and term to higher index and term, and the opposite order.